src/main/java/com/rocana/lucene/codec/v1/RocanaSearchCodecV1.java

```diff
@@ -32,15 +36,68 @@
  * the entire file improves Searcher open performance.
  *
  * If you have access to Rocana's JIRA, see ROCANA-8229.
+ *
+ * The idea behind this:
+ * 
+ * Ultimately we're trying to comment out one line of code in
+ * {@link RocanaBlockTreeTermsReader}'s constructor.
+ * 
+ * We don't want to customize Lucene's on-disk format. We just want
+ * to stop Lucene from checksumming an entire file when there's no
+ * benefit. The Lucene checksum is redundant since we store indexes
+ * on HDFS, which also checksums. The Lucene checksum happens at an
+ * inopportune time as we're trying to open a Rocana Search Searcher,
+ * which may happen during a rebalance, and we'd rather eliminate
+ * that penalty altogether or at least delay it until the rebalance
+ * finishes.
+ * 
+ * To accomplish that we register this class as the default Lucene
```

Comment:
We should probably also mention that this codec has its own unique "name" that will be written to the segment info files, rather than the Lucene54 name, even though we still use the Lucene54Codec under the hood for most operations.